### PR TITLE
T15173

### PR DESCRIPTION
**Pull request by mattdangerw**
https://phabricator.endlessm.com/T15173

### CODE REVIEW

File: tools/ka-customizer [hunk 1]
```diff
@@ -190,7 +190,7 @@ add_label('Choose a stock theme (overrides application theme):');
 let stock_theme_chooser = new Gtk.ComboBoxText({
     halign: Gtk.Align.START,
 });
-['', 'default', 'encyclopedia', 'news', 'buffet', 'preset_a', 'preset_b'].forEach((item) => {
+['', 'default', 'buffet', 'course', 'encyclopedia', 'escola', 'library_list', 'news', 'picard', 'preset_a', 'preset_b'].forEach((item) => {
```
Comment (tchx84) [inline tools/ka-customizer]: can we exclude picard theme here? that theme is not really meant for apps, and could cause confusion?

File: tools/ka-customizer-runner [hunk 2]
```diff
@@ -1,11 +1,14 @@
 #!/bin/bash -ex
 
+BRANCH=$1
```
Comment (tchx84) [inline tools/ka-customizer-runner]: is this script supposed to be run by another script or humans? if humans, then we should probably add some usage message?

File: tools/ka-customizer-runner [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+shift
+
 if [ "$1" = "update" ]; then
     shift
-    flatpak update com.endlessm.Platform master
-    flatpak update com.endlessm.Sdk master
+    flatpak update "com.endlessm.Platform//$BRANCH"
+    flatpak update "com.endlessm.Sdk//$BRANCH"
```
Comment (tchx84) [inline tools/ka-customizer-runner]: I got an error when running the ka-customizer:

```
+ BRANCH=
+ shift
+ '[' update = update ']'
+ shift
+ flatpak update com.endlessm.Platform//
Looking for updates...
Updating: com.endlessm.Platform/x86_64/eos3.1 from eos-runtimes
./ka-customizer-runner: line 11:  1703 Segmentation fault      (core dumped) flatpak update "com.endlessm.Platform//$BRANCH"
```

The segmentation fault part is probably unrelated, but I think the BRANCH is not getting replaced?

File: tools/ka-customizer-installer [hunk 4]
```diff
@@ -3,6 +3,9 @@
 APPDIR=$HOME/.local/share/applications
 BINDIR=$HOME/.local/bin
 
+if ! flatpak info --show-commit com.endlessm.Sdk eos3.1; then
+    flatpak install eos-runtimes com.endlessm.Sdk eos3.1
+fi
```
Comment (tchx84) [inline tools/ka-customizer-installer]: shouldn't we check for and include com.endlessm.Platform eos3.1 as well?

btw, we could also use a variable here ($BRANCH?) so we can easily replace eos3.1 later on (or even have a installer.in version that does it automatically)?

File: tools/ka-customizer-runner (hunk 2 above)
Comment (mattdangerw) [inline tools/ka-customizer-runner]: Nope, not intended for human invocation. ka-customizer runs this, and it handles chaining commands easier done in bash than gjs.

Presumably if you are a human and comfortable with flatpak and the command line, you can just use the flatpak and knowledge app cli options directly.

File: tools/ka-customizer-runner (hunk 3 above)
Comment (mattdangerw) [inline tools/ka-customizer-runner]: BRANCH is intentionally left blank in that case, to run the default branch for the app. I don't think there's an easy way to explicitly get the platform version a flatpak app has been built against, so its not easy to write the branch out explicitly at the moment. See discussion here https://phabricator.endlessm.com/T15173#321665

File: tools/ka-customizer-installer (hunk 4 above)
Comment (mattdangerw) [inline tools/ka-customizer-installer]: Can check for Platform sure. This is meant to be run on non converted eos images by design where it should already be installed, but can't hurt to check.

I can add a variable here, though honestly I'm not sure what this looks like come 3.2. Do we need to make sure all of 3.1, 3.2 and master are installed? Will some apps still be targeting the 3.1 runtime? Kinda suspect so, but I'm not totally worried about future proofing this quite yet.

**Pull request merged**